src/sql/src/plan/statement/ddl.rs

```diff
@@ -3494,14 +3495,40 @@ pub fn plan_create_connection(
             }
             Connection::AwsPrivatelink(connection)
         }
-        CreateConnection::Ssh { with_options } => {
-            let c = SshConnectionOptionExtracted::try_from(with_options)?;
+        CreateConnection::Ssh { options } => {
+            let c = SshConnectionOptionExtracted::try_from(options)?;
             let connection = mz_storage_client::types::connections::SshConnection::try_from(c)?;
             Connection::Ssh(connection)
         }
     };
     let name = scx.allocate_qualified_name(normalize::unresolved_item_name(name)?)?;
 
+    let mut user_validate = None;
+    for option in with_options {
+        match option.name {
+            CreateConnectionOptionName::Validate => {
+                scx.require_feature_flag(&vars::ENABLE_CONNECTION_VALIDATION_SYNTAX)?;
+                if user_validate.is_some() {
+                    sql_bail!("Option VALIDATE must be set at most once");
+                }
+                user_validate = match option.value {
+                    Some(WithOptionValue::Value(Value::Boolean(new_val))) => Some(new_val),
+                    None => Some(true),
+                    _ => sql_bail!("Option VALIDATE must be set to either TRUE or FALSE"),
+                };
+            }
+        }
+    }
```

Comment:
We have a macro that handles this sort of validation now! Grep around for `generate_extracted_config!`.

Reply:
Oh cool! Fixed